src/ray/gcs/actor_state_accessor.cc

```diff
@@ -0,0 +1,79 @@
+#include "ray/gcs/actor_state_accessor.h"
+#include <boost/none.hpp>
+#include "ray/gcs/redis_gcs_client.h"
+#include "ray/util/logging.h"
+
+namespace ray {
+
+namespace gcs {
+
+ActorStateAccessor::ActorStateAccessor(RedisGcsClient &client_impl)
+    : client_impl_(client_impl) {}
+
+Status ActorStateAccessor::AsyncGet(const JobID &job_id, const ActorID &actor_id,
+                                    const MultiItemCallback<ActorTableData> &callback) {
+  RAY_DCHECK(callback != nullptr);
+  auto on_done = [callback](RedisGcsClient *client, const ActorID &actor_id,
+                            const std::vector<ActorTableData> &data) {
+    callback(Status::OK(), data);
+  };
+
+  ActorTable &actor_table = client_impl_.actor_table();
+  return actor_table.Lookup(job_id, actor_id, on_done);
+}
+
+Status ActorStateAccessor::AsyncAdd(const JobID &job_id, const ActorID &actor_id,
```

Comment:
I think we should make the new GCS client API as precise as possible. 
That's why I suggest splitting this method into `AsyncAdd` and `AsyncUpdateState`. So we can prevent users accidentally change immutable fields, and also hide as much details in GCS as possible. 

Maybe we don't even have to expose `ActorTableData` in the API. For example, we can just pass actor creation task as the parameter of `AsyncAdd`, and compute `ActorTableData` internally. 

That might require massive change. We can that in follow-up PRs.

But in this PR, let's at least rename `AsyncAdd` to `AsyncAddOrUpdate`?

Reply:
> I think we should make the new GCS client API as precise as possible.
> That's why I suggest splitting this method into `AsyncAdd` and `AsyncUpdateState`. So we can prevent users accidentally change immutable fields, and also hide as much details in GCS as possible.
> 
> Maybe we don't even have to expose `ActorTableData` in the API. For example, we can just pass actor creation task as the parameter of `AsyncAdd`, and compute `ActorTableData` internally.
> 
> That might require massive change. We can that in follow-up PRs.
> 
> But in this PR, let's at least rename `AsyncAdd` to `AsyncAddOrUpdate`?

I also think that updates and additions should be two different interfaces. But ActorTable didn't distinguish those two semantics, so it's a little tricky to implemented in the upper layer.

Reply:
I think we can set `log_length = 0` in `AsyncAdd` and compute log_length based on state and remaining reconstructions in `AsyncUpdate`?

Reply:
How about add interface AsyncUpdate(maybe has the same params as AsyncAdd). In future, we can fix the params, and also we can overloading it to implement different update functions.

Reply:
> I think we can set `log_length = 0` in `AsyncAdd` and compute log_length based on state and remaining reconstructions in `AsyncUpdate`?

I think `remaining reconstructions` in redis module is better.

Reply:
ok, sounds good